editors/code/src/config.ts

```diff
@@ -98,96 +92,13 @@ export class Config {
-     * downloaded by the extension).
-     */
-    get prebuiltServerFileName(): null | string {
-        // See possible `arch` values here:
-        // https://nodejs.org/api/process.html#process_process_arch
-
-        switch (process.platform) {
-
-            case "linux": {
-                switch (process.arch) {
-                    case "arm":
-                    case "arm64": return null;
-
-                    default: return "rust-analyzer-linux";
-                }
-            }
-
-            case "darwin": return "rust-analyzer-mac";
-            case "win32": return "rust-analyzer-windows.exe";
-
-            // Users on these platforms yet need to manually build from sources
-            case "aix":
-            case "android":
-            case "freebsd":
-            case "openbsd":
-            case "sunos":
-            case "cygwin":
-            case "netbsd": return null;
-            // The list of platforms is exhaustive (see `NodeJS.Platform` type definition)
-        }
-    }
-
-    get installedExtensionUpdateChannel(): UpdatesChannel {
-        return this.extensionReleaseTag === NIGHTLY_TAG
-            ? UpdatesChannel.Nightly
-            : UpdatesChannel.Stable;
-    }
-
-    get serverSource(): null | ArtifactSource {
-        const serverPath = RA_LSP_DEBUG ?? this.serverPath;
-
-        if (serverPath) {
-            return {
-                type: ArtifactSource.Type.ExplicitPath,
-                path: Config.replaceTildeWithHomeDir(serverPath)
-            };
-        }
-
-        const prebuiltBinaryName = this.prebuiltServerFileName;
-
-        if (!prebuiltBinaryName) return null;
-
-        return this.createGithubReleaseSource(
-            prebuiltBinaryName,
-            this.extensionReleaseTag
-        );
-    }
-
-    private createGithubReleaseSource(file: string, tag: string): ArtifactSource.GithubRelease {
-        return {
-            type: ArtifactSource.Type.GithubRelease,
-            file,
-            tag,
-            dir: this.ctx.globalStoragePath,
-            repo: {
-                name: "rust-analyzer",
-                owner: "rust-analyzer",
-            }
-        };
-    }
-
-    get nightlyVsixSource(): ArtifactSource.GithubRelease {
-        return this.createGithubReleaseSource("rust-analyzer.vsix", NIGHTLY_TAG);
-    }
+    get globalStoragePath(): string { return this.ctx.globalStoragePath; }
 
     // We don't do runtime config validation here for simplicity. More on stackoverflow:
     // https://stackoverflow.com/questions/60135780/what-is-the-best-way-to-type-check-the-configuration-for-vscode-extension
 
-    private get serverPath() { return this.cfg.get("serverPath") as null | string; }
-    get updatesChannel() { return this.cfg.get("updates.channel") as UpdatesChannel; }
+    get serverPath() { return this.cfg.get("serverPath") as null | string; }
+    get channel(): "stable" | "nightly" { return this.cfg.get("updates.channel", "stable"); }
```

Comment:
See my comments on the outdated code.
And there is no need to specify the default `stable`. This is what I tried to eliminate in #3131, the second source of truth!